src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogAgentClient.java

```diff
@@ -734,63 +523,68 @@ public boolean sendLogs(String payload) {
-            DatadogUtilities.severe(logger, e, "Failed to start build trace");
-            reinitializeAgentHttpClient(true);
-            return false;
+    /**
+     * Posts a given payload to the Agent EVP Proxy, so it is forwarded to the Webhook Intake.
+     */
+    private void sendSpansToWebhook(List<net.sf.json.JSONObject> spans) {
+        for (net.sf.json.JSONObject span : spans) {
+            // webhook intake does not support batch requests
+            postWebhook(span.toString());
         }
     }
 
+    /**
+     * Posts a given payload to the Agent EVP Proxy, so it is forwarded to the Webhook Intake.
+     */
+    private void postWebhook(String payload) {
+        logger.fine("Sending webhook");
 
-    @Override
-    public boolean finishBuildTrace(BuildData buildData, Run<?, ?> run) {
-        try {
-            boolean status = reinitializeAgentHttpClient(false);
-            if(!status) {
-                return false;
-            }
+        DatadogGlobalConfiguration datadogGlobalDescriptor = DatadogUtilities.getDatadogGlobalDescriptor();
+        String urlParameters = datadogGlobalDescriptor != null ? "?service=" + datadogGlobalDescriptor.getCiInstanceName() : "";
+        String url = String.format("http://%s:%d/evp_proxy/v1/api/v2/webhook/%s", hostname, traceCollectionPort, urlParameters);
 
-            checkEvpProxySupportAndUpdateLogic();
+        Map<String, String> headers = new HashMap<>();
+        headers.put("X-Datadog-EVP-Subdomain", "webhook-intake");
+        headers.put("DD-CI-PROVIDER-NAME", "jenkins");
 
-            logger.fine("Finished build trace");
-            this.traceBuildLogic.finishBuildTrace(buildData, run);
-            return true;
-        } catch (Exception e) {
-            DatadogUtilities.severe(logger, e, "Failed to finish build trace");
-            reinitializeAgentHttpClient(true);
-            return false;
-        }
+        byte[] body = payload.getBytes(StandardCharsets.UTF_8);
+        client.postAsynchronously(url, headers, "application/json", body);
     }
 
-    @Override
-    public boolean sendPipelineTrace(Run<?, ?> run, FlowNode flowNode) {
+    private void sendSpansToApm(List<net.sf.json.JSONObject> spans) {
         try {
-            boolean status = reinitializeAgentHttpClient(false);
-            if(!status) {
-                return false;
+            Map<String, net.sf.json.JSONArray> tracesById = new HashMap<>();
+            for (net.sf.json.JSONObject span : spans) {
+                tracesById.computeIfAbsent(span.getString(JsonTraceSpanMapper.TRACE_ID), k -> new net.sf.json.JSONArray()).add(span);
             }
 
-            checkEvpProxySupportAndUpdateLogic();
+            final JSONArray jsonTraces = new JSONArray();
+            for(net.sf.json.JSONArray trace : tracesById.values()) {
+                jsonTraces.put(trace);
+            }
+            byte[] payload = jsonTraces.toString().getBytes(StandardCharsets.UTF_8);
```

Comment:
Is there a size limit in bytes we need to check to split the payload in different batches before sending it?